core/node/node_framework/src/implementations/layers/house_keeper.rs

```diff
@@ -196,8 +196,8 @@ impl Task for L1BatchMetricsReporterTask {
         "l1_batch_metrics_reporter"
     }
 
-    async fn run(self: Box<Self>, _stop_receiver: StopReceiver) -> anyhow::Result<()> {
-        self.l1_batch_metrics_reporter.run().await
+    async fn run(self: Box<Self>, stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        self.l1_batch_metrics_reporter.run(stop_receiver.0).await
```

Comment:
Huh, looks like you've resolved [QIT-29](https://linear.app/matterlabs/issue/QIT-29). Could you please remove the TODO above? BTW, thanks for that :)